### PR TITLE
FIX: Make font style- and stretch- check case insensitive

### PR DESCRIPTION
**Pull request by achabotl**
This makes style and stretch case insensitive in
`kiva.fonttools.font_manager.ttfFontProperty`. The case sensitivity led
to font styles not being properly detected on (at least) macOS.

The checks on filename were already case insensitive.

I'm not sure how to test this. There's a test in `fonttools`, but it would required modifying the test `.ttf` file and I didn't know how to fix it. Help would be appreciated.

https://github.com/enthought/enable/blob/6cf525cbeefdf34dc6eac5e54d7a73f755ea79b1/kiva/fonttools/tests/test_font_manager.py#L53

Here's a small test sample and an example that it works
```python
from kiva.fonttools import Font
from kiva.image import GraphicsContext
from kiva.constants import ITALIC, BOLD

gc = GraphicsContext((200, 200))

f = Font('', size=24)
print("Top: ", f.findfont())
gc.set_font(f)
gc.set_text_position(30, 150)
gc.show_text("Hello World")

f = Font('Georgia', size=24)
print('2nd', f.findfont())
gc.set_font(f)
gc.set_text_position(30, 110)
gc.show_text("Hello World")

f = Font('Georgia', size=24, style=ITALIC)
print('3rd', f.findfont())
gc.set_font(f)
gc.set_text_position(30, 70)
gc.show_text("Hello World")

f = Font('Georgia', size=24, style=BOLD)
print('4th', f.findfont())
gc.set_font(f)
gc.set_text_position(30, 30)
gc.show_text("Hello World")

gc.save('text.png')
```

text

Fixes: #404
Also addresses part of #391.

### CODE REVIEW

**Comment by achabotl** [comment]
The test failures are unrelated to the change.  


- Environment: RUNTIME=3.5, TOOLKITS=null pyqt pyqt5: `NotImplementedError: the null pyface.ui.null backend doesn't implement util.gui_test_assistant:GuiTestAssistant`
- Environment: RUNTIME=3.6, TOOLKITS=null pyqt pyqt5: `NotImplementedError: the null pyface.ui.null backend doesn't implement util.gui_test_assistant:GuiTestAssistant`
- Environment: RUNTIME=3.6, TOOLKITS=wx: `RuntimeError: Python 3.6, toolkit wx, and pillow pillow not supported by test environments`

**Comment by kitchoi** [comment]
Rebuilding CI now that #424 is merged... 🤞

**Comment by kitchoi** [comment]
(Merged master in to get the latest CI changes)

File: kiva/fonttools/tests/test_font_manager.py [hunk 1]
```diff
@@ -70,3 +70,29 @@ def test_font(self):
...
+        exp_name = "Test TTF"
+        exp_style = "italic"
+        exp_variant = "normal"
+        exp_weight = 400
+        exp_stretch = "normal"
+        exp_size = "scalable"
+
+        # When
+        font = TTFont(test_font)
+        print(getPropDict(font))
```
Comment (jwiggins) [inline kiva/fonttools/tests/test_font_manager.py:89]: This `print` should probably be removed, no?

File: kiva/fonttools/tests/test_font_manager.py [hunk 2]
```diff
@@ -70,3 +70,29 @@ def test_font(self):
...
+        entry = ttfFontProperty(test_font, TTFont(test_font))
```
Comment (jwiggins) [inline kiva/fonttools/tests/test_font_manager.py:90]: Why not `ttfFontProperty(test_font, font)`?

**Review by jwiggins — APPROVED** [review]
LGTM

**Comment by achabotl** [comment]
Thanks for the fix @kitchoi, sorry for not being responsive on this one. I had forgotten about this PR. 😞

**Comment by kitchoi** [comment]
@achabotl  No worries, thank you for the fix!

**Pull request merged**